api/include/opentelemetry/context/context_value.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <cstdint>
+
+#include "opentelemetry/nostd/span.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/nostd/variant.h"
+#include "opentelemetry/version.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace context
+{
+using ContextValue = nostd::variant<bool,
```

Comment:
Please also consider [this comment](https://github.com/open-telemetry/opentelemetry-cpp/pull/180#discussion_r456120855).

Reply:
When I try to add nostd::unique_ptr<char []>.
I get 'opentelemetry::v0::nostd::variant<Ts>::variant(const opentelemetry::v0::nostd::variant<Ts>&) [with Ts = {bool, long int, 
long unsigned int, double, opentelemetry::v0::nostd::unique_ptr<char []>}]' is implicitly deleted because the default definition would be ill-formed:
   variant(const variant &) = default;

Reply:
and the same for nostd::shared_ptr<SpanContext>. Do you have any idea why that is?